src/main/java/nl/praegus/fitnesse/responders/allTags/AllTagsResponder.java

```diff
@@ -0,0 +1,77 @@
+package nl.praegus.fitnesse.responders.allTags;
+
+import fitnesse.FitNesseContext;
+import fitnesse.authentication.SecureOperation;
+import fitnesse.authentication.SecureReadOperation;
+import fitnesse.authentication.SecureResponder;
+import fitnesse.http.Request;
+import fitnesse.http.Response;
+import fitnesse.http.SimpleResponse;
+import fitnesse.wiki.*;
+import fitnesse.wikitext.parser.SourcePage;
+import org.json.JSONObject;
+
+import java.io.UnsupportedEncodingException;
+import java.util.*;
+
+public class AllTagsResponder implements SecureResponder {
+    private JSONObject tagObject = new JSONObject();
+
+    @Override
+    public Response makeResponse(FitNesseContext fitNesseContext, Request request) throws Exception {
+        WikiSourcePage sourcePage = new WikiSourcePage(loadPage(fitNesseContext, request.getResource(), request.getMap()));
+
+        return makeTagResponse(sourcePage);
+    }
+
+    public SimpleResponse makeTagResponse(SourcePage sourcePage) throws UnsupportedEncodingException {
+        tagObject.put("Tags", getPageTags(sourcePage, new HashSet<>()));
+
+        SimpleResponse response = new SimpleResponse();
+        response.setMaxAge(0);
+        response.setStatus(200);
+        response.setContentType("application/json");
+        response.setContent(tagObject.toString(3));
+        return response;
+    }
+
+    public Set<String> getPageTags(SourcePage page, Set<String> allTags){
+        String[] tags = page.getProperty(WikiPageProperty.SUITES).split("\\s*,\\s*");
+        for(String tag: tags) {
+            if (tag.length() > 0) {
+                allTags.add(tag);
+            }
+        }
+
+        for (SourcePage p : getSortedChildren(page)) {
+            getPageTags(p, allTags);
+        }
+
+        return allTags;
+    }
+
+    @Override
+    public SecureOperation getSecureOperation() {
+        return new SecureReadOperation();
+    }
+
+    private List<SourcePage> getSortedChildren(SourcePage parent) {
```

Comment:
waarom sorteren we die dingen?

Reply:
De sorting is weggehaald. Sorteren is onnodig voor deze responder, aangezien het niet uitmaakt in welke volgorde de tags zijn.